### PR TITLE
Fix Configuration Overview and API reference docs

### PR DESCRIPTION
**Pull request by prudrabhat**
- Add annotations
- Fix links
- Remove references to older Android versions

## Description
Configuration Overview and API reference docs have a few broken links and references to older versions. Remove them.

## Related Issue
- #199
-  #198 

## How Has This Been Tested?
- Run `yarn dev` locally and verify links and pages.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by cdhoffmann — APPROVED** [review]

File: src/pages/documentation/mobile-core/configuration/api-reference.md [hunk 1]
```diff
@@ -62,7 +56,7 @@ Android
 
 ## configureWithFileInPath
 
-You can include a bundled JSON configuration file in your app package to replace or complement the configuration that was downloaded by using the [Configure with App ID per environment](./index.md#configure-with-app-id-per-environment) approach.
+You can include a bundled JSON configuration file in your app package to replace or complement the configuration that was downloaded by using the [Configure with App ID per environment](#configure-with-app-id-per-environment) approach.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/configuration/api-reference.md:59]: I don't know if this link update is correct? Isn't the "Configure with App ID per environment" listed on the overview page? Not in the API?

**Review by calebk1 — COMMENTED** [review]
One question, but looks good otherwise.

File: src/pages/documentation/mobile-core/configuration/api-reference.md (hunk 1 above)
Comment (prudrabhat) [reply src/pages/documentation/mobile-core/configuration/api-reference.md]: Whoops. Yes. You are right. This should be `.#configure-with-app-id-per-environment`. I incorrectly used the version from Overview.md

File: src/pages/documentation/mobile-core/configuration/api-reference.md [hunk 2]
```diff
@@ -48,11 +46,7 @@ iOS
 
 ## configureWithFileInAssets
 
-<InlineAlert variant="info" slots="text"/>
-
-This API is only available in Android and was added in Android was added in Android Core version 1.7.0.
-
-You can bundle a JSON configuration file in the app's Assets folder to replace or complement the configuration that was downloaded by using the [Configure with App ID per environment](./index.md#configure-with-app-id-per-environment) approach.
+You can bundle a JSON configuration file in the app's Assets folder to replace or complement the configuration that was downloaded by using the [Configure with App ID per environment](../#configure-with-app-id-per-environment) approach.
```
Comment (calebk1) [inline src/pages/documentation/mobile-core/configuration/api-reference.md:49]: Actually just to confirm - is this a link within the Configuration overview page? There's some... wonkiness with relative links within tabs, which is why I want to confirm that before giving approval. 

(Basically, relative links within tabs are treated as links within the document itself, so the relative link won't work on VSCode, but it _does_ work on the site. It's weird.)

**Review by calebk1 — COMMENTED** [review]
Sorry! One more comment/question.

File: src/pages/documentation/mobile-core/configuration/api-reference.md (hunk 2 above)
Comment (prudrabhat) [reply src/pages/documentation/mobile-core/configuration/api-reference.md:49]: Yes. This is a link in the Configuration overview page. I tested this running an instance locally and verified that it does resolve to the right location. I can also change this to `../index.md#configure-with-app-id-per-environment`  which is equivalent.

Comment (calebk1) [reply src/pages/documentation/mobile-core/configuration/api-reference.md:49]: Oooh okay - so long as you ran it locally and it works, that's all fine with me.

**Pull request merged**